pam/internal/gdm/protocol_test.go

```diff
@@ -623,3 +627,76 @@ func TestGdmStructsUnMarshal(t *testing.T) {
 		})
 	}
 }
+
+func TestSafeString(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		eventData *gdm.EventData
+
+		wantString     string
+		wantSafeString string
+	}{
+		"Empty_gdm_data_is_stringified": {
+			eventData: &gdm.EventData{},
+		},
+		"Non-AuthenticatedRequest_is_fully_stringified": {
+			eventData: &gdm.EventData{
+				Type: gdm.EventType_authEvent,
+				Data: &gdm.EventData_AuthEvent{
+					AuthEvent: &gdm.Events_AuthEvent{
+						Response: &authd.IAResponse{
+							Access: auth.Granted,
+							Msg:    "Hello!",
+						},
+					},
+				},
+			},
+			wantString: `type:authEvent authEvent:{response:{access:"granted" msg:"Hello!"}}`,
+		},
+		"AuthenticatedRequest_without_secret_is_fully_stringified": {
```

Comment:
Is this test case misnamed? The way I understand it, it does have a secret and it tests that the request is censored in the string.

Reply:
Sorry I meant to add another test which I didn't, added a fixup.